appinventor/components/src/com/google/appinventor/components/runtime/ButtonBase.java

```diff
@@ -169,16 +169,22 @@ public boolean onTouch(View view, MotionEvent me) {
       if (me.getAction() == MotionEvent.ACTION_DOWN) {
         //button pressed, provide visual feedback AND return false
         if (ShowFeedback() && (AppInventorCompatActivity.isClassicMode() || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP)) {
-          view.getBackground().setAlpha(70); // translucent
-          view.invalidate();
+          if (backgroundImageDrawable != null || imagePath != "") {
```

Comment:
Rather than checking `imagePath != ""`, which isn't guaranteed to be true even if `imagePath` is empty (`==` checks identity), you should do `imagePath.length() > 0`, but if this is true then the backgroundImageDrawable shouldn't be null.